src/middlewared/middlewared/etc_files/hostname.py

```diff
@@ -7,4 +7,8 @@
 
 async def render(service, middleware):
     config = await middleware.call("network.configuration.config")
-    await run(["hostname", f'{config["hostname_local"]}.{config["domain"]}'])
+    hostname = f"{config['hostname_local']}.{config['domain']}"
+    with open('/etc/hostname', 'w') as f:
```

Comment:
Since we're doing IO now, I guess this can block event loop so probably should make it synchronous or run in thread?